sway-core/src/language/ty/declaration/function.rs

```diff
@@ -182,12 +180,95 @@ impl MaterializeConstGenerics for TyFunctionDecl {
     }
 }
 
+/// Rename cosnt generics when the name inside the struct/enum does not match
+/// the name in the impl items.
```

Comment:
Just for preciseness of the comment, because the const generic name will actually be in the `impl<...>`, but it will be _used_ in impl items. So it's either "name in the impl" or, e.g. "name used in the impl items".  
```suggestion
/// the name in the impl.
```